### DISCUSSION TITLE
How to get directory by DirectorySearch?

### DISCUSSION THREAD

**Discussion by MasterBran** [wixtoolset/issues#6791] [Questions] [answered]
For some reason, I need to install some file to the `XLSTART` folder under the Excel's path, first I get the file path by this code 
`<Property Id="OFFICEPATH">
            <RegistrySearch Id="OFFICE" Type="file" Root="HKLM" Key="SOFTWARE\Microsoft\Windows\CurrentVersion\App Paths\excel.exe" Name="InstallDir" />
</Property>
`
and then, I use 'DirectorySearch' to get the folder path of  `excel.exe`(Something wrong here).
`
<Property Id="OFFICEDIR">
            <DirectorySearch Id="EXCELDIR" Path="[OFFICEPATH]" Depth="0" AssignToProperty="yes">
                <FileSearch Id="EXCELXE" Name="excel.exe" />
            </DirectorySearch>
</Property>
<SetDirectory Id="OFFICETARGETDIR" Value="[OFFICEDIR]" />
`
Finally, I specify the target path like this 
`
<Directory Id="TARGETDIR" Name="SourceDir">
            <Directory Id="OFFICETARGETDIR">
                <Directory Id="INSTALLFOLDER" Name="XLSTART" />
            </Directory>
</Directory>
`
but it's not worked, all of my file installed in `F:\XLSTART`. How could I do this?

**Comment by MasterBran** [comment]
Finally I found the bug and resolve it.

**Reply by robmen** [reply]
That's good news, @MasterBran. Could you maybe post your solution here and mark it as the answer so if others with similar issues find this thread in the future they'll know what you did to solve it?

**Reply by MasterBran** [reply]
Thanks for replay, when I realized that was very easy problem, I thought most people might not need it,so...
Anyway, I post my solution and marked as answer.

**Comment by MasterBran** [comment, marked as answer]
Maybe that is actually a very simple but possibly overlooked problem, the bug occur in `<Property Id="OFFICEPATH"> <RegistrySearch Id="OFFICE" Type="file" Root="HKLM" Key="SOFTWARE\Microsoft\Windows\CurrentVersion\App Paths\excel.exe" Name="InstallDir" /> </Property>`,the attribute `Name`  of `RegistrySearch`  is wrong, that must same to the registry item name(In my case,there should have no attribute `Name`), and then I could get the folder path of `excel.exe`. The `DirectorySearch` is needless, so this is my final code
`<Property Id="OFFICEPATH"> <RegistrySearch Id="OFFICE" Type="file" Root="HKLM" Key="SOFTWARE\Microsoft\Windows\CurrentVersion\App Paths\excel.exe" Name="InstallDir" /> </Property>
<SetDirectory Id="OFFICETARGETDIR" Value="[OFFICEDIR]" />
<Directory Id="TARGETDIR" Name="SourceDir"> <Directory Id="OFFICETARGETDIR"> <Directory Id="INSTALLFOLDER" Name="XLSTART" /> </Directory> </Directory>
`
